library/kotlin/src/io/envoyproxy/envoymobile/StreamCallback.kt

```diff
@@ -0,0 +1,55 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+interface StreamCallback {
+  /**
+   * Invoked whenever the response headers have been read.
+   *
+   * @param headers the headers of the response.
+   * @param statusCode the status code of the response.
+   */
+  fun onHeaders(headers: Map<String, List<String>>, statusCode: Int)
+
+  /**
+   * Invoked whenever the response body is read.
```

Comment:
it might not be the case that all response body gets sent back at once. i.e there might be multiple onData callbacks fired for a single stream.

In general I would say `Called when a data frame is received by the stream`. Similar in the other function comments here.

Reply:
I can make this change @rebello95 WDYT?

Reply:
Sure, SGTM